### ISSUE TITLE
Sporatic downloads on Linux

### ISSUE THREAD

**Issue by Sbackus65** [rmcrackan/Libation#717]
**Describe the bug**
After scanning the new library, only about every seventh or eighth book will decrypt.  It appears to fully download, then states error and moves to next audiobook

**To Reproduce**
1.  Scan library
2. Under pull down menu "Visible Books" select "Liberate:"
3. See errors as books appear to fully download, then state Error, will retry later

**Expected behavior**
Item number 3 above should be showing each book as downloaded.

**Screenshots**
![image](https://github.com/rmcrackan/Libation/assets/113722306/0655dfc5-dbbf-45b1-92df-8cf9481b2d07)

**Platform**

Linux Kubuntu 23.04, with system update yesterday; Libation 11.0.1

**Log Files**

**Comment by Sbackus65** [comment]
[_Log202308 (1).zip](https://github.com/rmcrackan/Libation/files/12375582/_Log202308.1.zip)

**Comment by Mbucari** [comment]
Libation is failing to move your audiobooks from the temp folder at `/tmp/Libation/DecryptInProgress/` to `/media/sbackus/Media/Audiobooks/Audible`. Does Libation have write access to that location? Is there enough space?

**Comment by Sbackus65** [comment]
There was enough space on both drives.  There is stilll 15.5TB free on the Media drive but the temp folder has now filled the remainder of the main drive.  

Libation should have write access to the Audible folder, as several books have been decrypted and moved to that location.  The books that are not decrypted have empty folders at that location.  Is it possible that some books do not require decryption? or some other step in the process?  Maybe it is a timing issue?

**Comment by Sbackus65** [comment]
Interesting.  I tried to set up a tmp folder in the media drive and do not seem able to move some of the tmp items in the main hard drive to this folder.  Error message is: "Could not write to /media/sbackus/Media/Audiobooks/tmp/..."  If I hit skip it skips some of the files...  Checking my permissions I seem to have all permissions.  Is it possible with the path some of the file names are too long for the file system?

**Comment by Sbackus65** [comment]
You are right!  All of the files in the /tmp/Libation/ subfolders have colons.  I do not see any files that were successfully moved that have colons.  Why would the colons be acceptable on the main drive but not on an external hard drive?

**Comment by Sbackus65** [comment]
Yes, external drive is NTFS.  Should I reformat as ExFat?

**Comment by Sbackus65** [comment]
Or another format? EXT4?

**Comment by Mbucari** [comment]
Either do that, or make use of Libation's character replacement settings.

![image](https://github.com/rmcrackan/Libation/assets/37587114/cd07ecf4-207c-4632-9d35-bd614a86a832)


The defaults are different on Linux and Windows because windows is always on NTFS, so you'd need to manually add the illegal characters to the list.

**Comment by Mbucari** [comment]
But this actually is a bug in Libation. It should check for illegal path characters on the destination drive.

**Comment by Mbucari** [comment]
If you do reformat, choose ext4. exfat extend FAT which was also developed by Microsoft and has the same illegal characters.

**Comment by Sbackus65** [comment]
ext4 it is! :-)

**Comment by Sbackus65** [comment]
Reformat to ext4 and these problems all just went away.  I could not get Libation to start again, so I deleted the Libation folder and ran again as a new installation.  That fixed the start issue.  Runs perfectly now.  Thank you again!
